### DISCUSSION TITLE
How to install libinjection ?

### DISCUSSION THREAD

**Discussion by sadicann** [corazawaf/coraza#107] [Q&A] [answered]
Hey,
How can i install and link libinjection ? I searched a bit but couldn't understand.

**Comment by jptosso** [comment, marked as answer]
Hey @sadicann, libinjection is not mandatory but you can install with the following procedure:

```sh
		git clone https://github.com/libinjection/libinjection
		gcc -std=c99 -Wall -Werror -fpic -c libinjection/src/libinjection_sqli.c -o libinjection/libinjection_sqli.o 
		gcc -std=c99 -Wall -Werror -fpic -c libinjection/src/libinjection_xss.c -o libinjection/libinjection_xss.o
		gcc -std=c99 -Wall -Werror -fpic -c libinjection/src/libinjection_html5.c -o libinjection/libinjection_html5.o
		gcc -dynamiclib -shared -o libinjection/libinjection.so libinjection/libinjection_sqli.o libinjection/libinjection_xss.o libinjection/libinjection_html5.o
		cp libinjection/*.so /usr/local/lib
		cp libinjection/*.o /usr/local/lib
		cp libinjection/src/*.h /usr/local/include/
		chmod 444 /usr/local/include/libinjection*
		ldconfig
```
In case of windows it is not compatible yet and in case of OS X you must change .so to .dylib. Also in some cases you must type ```ldconfig /usr/local/lib```.

Note in case you don't /usr/local/include nor /usr/local/lib, you may have to set env variable ```CGO_LDFLAGS```
There is also this article: https://coraza.io/docs/tutorials/dependencies/

Please let me know if you succeed.

**Comment by sadicann** [comment]
Thank you. I searched a lot for this, I had not seen it on the website. 
It would be great if there is a step-by-step tutorial for the installation.

**Reply by jptosso** [reply]
You are welcome
I agree with you but on Coraza v2 libinjection compatibility will be deprecated and it will run as an external plugin, I'm also working on a Golang implementation of libinjection but it doesn't work yet: https://github.com/jptosso/libinjection-go

Once the golang implementation is ready I will patch v1 and v2.

**Reply by sadicann** [reply]
I congratulate you for the work you do. I would like to contribute v2, but I do not have enough knowledge. I wish you continued work.

**Reply by jptosso** [reply]
Thank you very much @sadicann!

There are many ways to contribute to Coraza, testing and writing tests is a great way to start. Also there are hundreds of TODOs that aren't to hard to fix, it just takes time. You can also write plugins, it's easy and adds a lot of value to the project.
